Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -15857,6 +15858,17 @@ void WebPageProxy::setPresentingApplicationAuditToken(const audit_token_t& prese
 }
 #endif
 
+#if ENABLE(CONTENT_EXTENSIONS)
+void WebPageProxy::didExeedFrameNetworkUsage(WebCore::FrameIdentifier frameID, String host)
+{
+    auto& throttler = WebCore::ResourceMonitorManager::singleton().throttler();
```

Comment:
Using a singleton here is fishy, odds are the object should be owned by the WebsiteDataStore. It seems risky to use the same data structure for regular browsing and private browsing for example. Similarly, when closing a private browsing tab, it seems wrong to leave data such as host names from the private browsing session in this shared data structure.